pkg/util/nativeutils/eventbus/mock.go

```diff
@@ -61,6 +61,60 @@ func CreateFrameStreamer(topic topics.Topic) (*EventBus, io.WriteCloser) {
 
 // SimpleStreamer is a WriteCloser
 var _ io.WriteCloser = (*SimpleStreamer)(nil)
+var _ io.WriteCloser = (*StupidStreamer)(nil)
+
+// StupidStreamer is a streamer meant for using when testing internal
+// forwarding of binary packets through the ring buffer. It does *not* add
+// magic, frames or other wraps on the forwarded packet
+type StupidStreamer struct {
+	*bufio.Reader
+	*bufio.Writer
+}
+
+// NewStupidStreamer returns an initialized SimpleStreamer.
+func NewStupidStreamer() *StupidStreamer {
+	r, w := io.Pipe()
+	return &StupidStreamer{
+		Reader: bufio.NewReader(r),
+		Writer: bufio.NewWriter(w),
+	}
+}
+
+// Write receives the packets from the ringbuffer and writes it on the internal
+// pipe immediatelyh
+func (sss *StupidStreamer) Write(p []byte) (n int, err error) {
+	lg := make([]byte, 4)
+	binary.LittleEndian.PutUint16(lg, uint16(len(p)))
```

Comment:
Is there a reason you're using a `uint16` here over a `uint32`? The `uint32` would utilize all bytes, and allow for testing with larger packets if needed in the future.

Reply:
Sure thing. Does not change much and would avoid subtle hard-to-detect bugs in case we would test with bulky packages. Will change